buildSrc/src/main/groovy/EhOsgi.groovy

```diff
@@ -0,0 +1,99 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.DefaultInheritManifest
+import groovy.json.JsonSlurper
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.ProjectDependency
+import org.gradle.api.internal.file.FileResolver
+import org.gradle.api.plugins.osgi.OsgiPluginConvention
+import scripts.Utils
+
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * EhOsgi
+ *   OSGI additions to the manifest controlled by osgi key in gradle.properties
+ *   This plugin supports shadowJar if available
+ */
+class EhOsgi implements Plugin<Project> {
+
+  @Override
+  void apply(Project project) {
+    def utils = new Utils(project.baseVersion, project.logger)
+    def hashsetOfProjects = project.configurations.compile.dependencies.withType(ProjectDependency).dependencyProject
+    hashsetOfProjects << project  //self also, in case the invoking project defines osgi properties
+
+    project.plugins.apply 'java'
+    project.plugins.apply 'maven'
+    project.plugins.apply 'signing'
+//    project.plugins.apply 'com.github.johnrengelman.shadow'
```

Comment:
Please, delete all the commented out lines that are not comments - applies to the whole PR